aligncheck does not support build tags

Note: The field http.Request.Cancel was added in go1.5 and does not exist in go 1.4

I have two versions of code that are gated on build flags for a version of go.  Aligncheck checks both versions.  The command `aligncheck .` should skip running on files that aren't for the current build env.

```
< aligncheck .
cancel15.go:11:2: invalid operation: req (variable of type *net/http.Request) has no field or method Cancel
2015/12/15 16:36:36 could not type check: couldn't load packages due to errors:
```

The file looks like this:

```
// +build go1.5

package sfxclient

import (
    "golang.org/x/net/context"
    "net/http"
)

func (h *HTTPDatapointSink) withCancel(ctx context.Context, req *http.Request) (err error) {
    req.Cancel = ctx.Done()
    return h.handleResponse(h.Client.Do(req))
}
```

--- a/cmd/aligncheck/aligncheck.go
+++ b/cmd/aligncheck/aligncheck.go
@@ -20,17 +20,22 @@ import (
 	"log"
 	"os"
 	"sort"
+	"strings"
 	"unsafe"
+
+	"go/types"
 
 	"github.com/kisielk/gotool"
 	"golang.org/x/tools/go/loader"
-	"go/types"
 )
 
-var stdSizes = types.StdSizes{
-	WordSize: int64(unsafe.Sizeof(int(0))),
-	MaxAlign: 8,
-}
+var (
+	stdSizes = types.StdSizes{
+		WordSize: int64(unsafe.Sizeof(int(0))),
+		MaxAlign: 8,
+	}
+	buildTags = flag.String("tags", "", "Build tags")
+)
 
 func main() {
 	flag.Parse()
@@ -42,6 +47,9 @@ func main() {
 	}
 
 	ctx := build.Default
+	if *buildTags != "" {
+		ctx.BuildTags = strings.Split(*buildTags, ",")
+	}
 	loadcfg := loader.Config{
 		Build: &ctx,
 	}

--- a/cmd/aligncheck/aligncheck.go
+++ b/cmd/aligncheck/aligncheck.go
@@ -48,7 +48,7 @@ func main() {
 
 	ctx := build.Default
 	if *buildTags != "" {
-		ctx.BuildTags = strings.Split(*buildTags, ",")
+		ctx.BuildTags = strings.Fields(*buildTags)
 	}
 	loadcfg := loader.Config{
 		Build: &ctx,

--- a/cmd/varcheck/varcheck.go
+++ b/cmd/varcheck/varcheck.go
@@ -24,13 +24,15 @@ import (
 	"sort"
 	"strings"
 
+	"go/types"
+
 	"github.com/kisielk/gotool"
 	"golang.org/x/tools/go/loader"
-	"go/types"
 )
 
 var (
 	reportExported = flag.Bool("e", false, "Report exported variables and constants")
+	buildTags      = flag.String("tags", "", "Build tags")
 )
 
 type object struct {
@@ -136,6 +138,9 @@ func main() {
 	}
 
 	ctx := build.Default
+	if *buildTags != "" {
+		ctx.BuildTags = strings.Split(*buildTags, ",")
+	}
 	loadcfg := loader.Config{
 		Build: &ctx,
 	}

--- a/cmd/varcheck/varcheck.go
+++ b/cmd/varcheck/varcheck.go
@@ -139,7 +139,7 @@ func main() {
 
 	ctx := build.Default
 	if *buildTags != "" {
-		ctx.BuildTags = strings.Split(*buildTags, ",")
+		ctx.BuildTags = strings.Fields(*buildTags)
 	}
 	loadcfg := loader.Config{
 		Build: &ctx,

